src/settings/PasswordForm.ts

```diff
@@ -1,281 +1,267 @@
-							   Dialog.message("passwordResetFailed_msg")
-						   })
-			showProgressDialog("pleaseWait_msg", p)
-		}
-
-		Dialog.showActionDialog({
-			title: lang.get("changePassword_label"),
-			child: form,
-			validator: () => form.getErrorMessageId(),
-			okAction: changeUserPasswordAsAdminOkAction,
-		})
+	private getErrorFromStatus(status: Status): TranslationKey | null {
+		if (!status) return null
+		return status.type !== "valid" ? status.text : null
 	}
 
-	_onOldPasswordInput(oldPassword: string): void {
-		this._oldPassword = oldPassword
-
-		if (this._validateOldPassword && oldPassword === "") {
-			this._oldPasswordStatus = {
+	getOldPasswordStatus(): Status {
+		if (this.checkOldPassword && this.oldPassword() === "") {
+			return {
 				type: "neutral",
 				text: "oldPasswordNeutral_msg",
 			}
 		} else {
-			this._oldPasswordStatus = {
+			return {
 				type: "valid",
 				text: "emptyString_msg",
 			}
 		}
 	}
 
-	_onNewPasswordInput(newPassword: string): void {
-		this._newPassword = newPassword
-
-		if (this._newPassword === "") {
-			this._newPasswordStatus = {
+	getNewPasswordStatus(): Status {
+		if (this.newPassword() === "") {
+			return {
 				type: "neutral",
 				text: "password1Neutral_msg",
 			}
-		} else if (this._validateOldPassword && this._oldPassword === this._newPassword) {
-			this._newPasswordStatus = {
+		} else if (this.checkOldPassword && this.oldPassword() === this.newPassword()) {
+			return {
 				type: "invalid",
 				text: "password1InvalidSame_msg",
 			}
-		} else if (this.isPasswordUnsecure()) {
-			if (this._enforcePasswordStrength) {
-				this._newPasswordStatus = {
+		} else if (this.isPasswordInsecure()) {
+			if (this.enforceStrength) {
+				return {
 					type: "invalid",
 					text: "password1InvalidUnsecure_msg",
 				}
 			} else {
-				this._newPasswordStatus = {
+				return {
 					type: "valid",
 					text: "password1InvalidUnsecure_msg",
 				}
 			}
 		} else {
-			this._newPasswordStatus = {
+			return {
 				type: "valid",
 				text: "passwordValid_msg",
 			}
 		}
 	}
 
-	_onRepeatedPasswordInput(repeatedPassword: string): void {
-		this._repeatedPassword = repeatedPassword
+	getRepeatedPasswordStatus(): Status {
+		const repeatedPassword = this.repeatedPassword()
+		const newPassword = this.newPassword()
 
-		if (this._repeatPassword && this._repeatedPassword === "") {
-			this._repeatedPasswordStatus = {
+		if (this.repeatInput && repeatedPassword === "") {
+			return {
 				type: "neutral",
 				text: "password2Neutral_msg",
 			}
-		} else if (this._repeatPassword && this._repeatedPassword !== this._newPassword) {
-			this._repeatedPasswordStatus = {
+		} else if (this.repeatInput && repeatedPassword !== newPassword) {
+			return {
 				type: "invalid",
 				text: "password2Invalid_msg",
 			}
 		} else {
-			this._repeatedPasswordStatus = {
+			return {
 				type: "valid",
 				text: "passwordValid_msg",
 			}
 		}
 	}
+
+	isPasswordInsecure(): boolean {
```

Comment:
Few of those functions are now called on every render. I wonder if it's doing too much work (especially in the sensitive places like `MailEditor`). Maybe we could only recalculate it on input (the model has state anyway so it's already a good fit)